main.tf

```diff
@@ -118,26 +159,93 @@ resource "ibm_is_instance" "vsi" {
     ]
   }
 
-  primary_network_interface {
-    subnet = each.value.subnet_id
-    security_groups = flatten([
-      (var.create_security_group ? [ibm_is_security_group.security_group[var.security_group.name].id] : []),
-      var.security_group_ids,
-      (var.create_security_group == false && length(var.security_group_ids) == 0 ? [data.ibm_is_vpc.vpc.default_security_group] : []),
-    ])
-    allow_ip_spoofing = var.allow_ip_spoofing
-    dynamic "primary_ip" {
-      for_each = var.manage_reserved_ips ? [1] : []
-      content {
-        reserved_ip = ibm_is_subnet_reserved_ip.vsi_ip[each.value.name].reserved_ip
+  # Primary Virtual Network Interface
+  dynamic "primary_network_attachment" {
+    for_each = var.use_legacy_network_interface ? [] : [1]
+    content {
+      name = "${each.value.subnet_name}-eth0"
+      virtual_network_interface {
+        subnet = each.value.subnet_id
+        security_groups = flatten([
+          (var.create_security_group ? [ibm_is_security_group.security_group[var.security_group.name].id] : []),
+          var.security_group_ids,
+          (var.create_security_group == false && length(var.security_group_ids) == 0 ? [data.ibm_is_vpc.vpc.default_security_group] : []),
+        ])
+        allow_ip_spoofing         = var.allow_ip_spoofing
+        auto_delete               = true
+        enable_infrastructure_nat = true
+        dynamic "primary_ip" {
+          for_each = var.manage_reserved_ips ? [1] : []
+          content {
+            reserved_ip = ibm_is_subnet_reserved_ip.vsi_ip[each.value.name].reserved_ip
+          }
+        }
+        dynamic "ips" {
+          for_each = var.number_of_secondary_reserved_ips > 0 && var.manage_reserved_ips ? { for count in range(var.number_of_secondary_reserved_ips) : count => count } : {}
+          content {
+            reserved_ip = ibm_is_subnet_reserved_ip.secondary_vsi_ip["${each.value.name}-${ips.key}"].reserved_ip
+          }
+        }
+      }
+    }
+  }
+
+  # Additional Virtual Network Interface
+  dynamic "network_attachments" {
+    for_each = {
+      for k in var.secondary_subnets : k.zone => k
+      if k.zone == each.value.zone && !var.use_legacy_network_interface
+    }
+    content {
+      name = network_attachments.value.name
+      virtual_network_interface {
```

Comment:
I do not see the primary (optional) reserved IP being assigned to the primary vni? I wasn't sure how the new VNI works, should it have an optional reserved IP entry?

Reply:
The primary reserved IP for the new VNI is being set at line number 177